dev-docs/lucene-upgrade.md

```diff
@@ -0,0 +1,93 @@
+<!--
+    Licensed to the Apache Software Foundation (ASF) under one or more
+    contributor license agreements.  See the NOTICE file distributed with
+    this work for additional information regarding copyright ownership.
+    The ASF licenses this file to You under the Apache License, Version 2.0
+    the "License"); you may not use this file except in compliance with
+    the License.  You may obtain a copy of the License at
+
+        http://www.apache.org/licenses/LICENSE-2.0
+
+    Unless required by applicable law or agreed to in writing, software
+    distributed under the License is distributed on an "AS IS" BASIS,
+    WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+    See the License for the specific language governing permissions and
+    limitations under the License.
+ -->
+
+# Lucene upgrade steps
+
+## Read
+
+https://lucene.apache.org/core/9_4_0/MIGRATE.html
+
+## Start
+
+Create a new branch locally e.g. `git checkout -b lucene940 -t origin/main` for upgrading to Lucene 9.4.0 version.
+
+## Build
+
+### `versions.props` update
+
+```
+- org.apache.lucene:*=9.3.0
++ org.apache.lucene:*=9.4.0
+```
+
+### `versions.lock` update
+
+```
+gradlew --write-locks
+```
+
+### `solr/licenses` update
+
+```
+git rm lucene-*-9.3.0.jar.sha1
```

Comment:
`./gradlew updateLicenses` will handle this

Reply:
thanks! #1087 to simplify the docs (and I'll leave that PR open for a few days in case of additional documentation feedback)